marine.c

```diff
@@ -633,7 +634,50 @@ WS_DLL_PUBLIC int validate_display_filter(char *dfilter) {
     return TRUE;
 }
 
-WS_DLL_PUBLIC int marine_add_filter(char *bpf, char *dfilter, char **fields, int fields_len, char *err_msg) {
+int parse_output_fields(output_fields_t *output_fields, char **fields, int fields_len, char **err_msg) {
+    for (int i = 0; i < fields_len; i++) {
+        output_fields_add(output_fields, fields[i]);
+    }
+
+    GSList *it = NULL;
+    GSList *invalid_fields = output_fields_valid(output_fields);
+    if (invalid_fields != NULL) {
+        if (err_msg == NULL) {
+            return -1;
+        }
+
+        int current_err_size = 64;
+        *err_msg = (char *)malloc(current_err_size);
```

Comment:
`*err_msg = strdup("Some fields aren't valid:\n")`
Will allocate and copy in one go :)